src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,188 @@
+     
+        # Load Conditional Data 
+        cbsd_conditional_device_1 = {
+           'cbsdCategory': device_1['cbsdCategory'],
+           'fccId': device_1['fccId'],
+           'cbsdSerialNumber': device_1['cbsdSerialNumber'],
+           'airInterface': device_1['airInterface'],
+           'installationParam': device_1['installationParam'],
+           'measCapability': device_1['measCapability']
+         }
+        cbsd_conditional_device_2 = {
+           'cbsdCategory': device_2['cbsdCategory'],
+           'fccId': device_2['fccId'],
+           'cbsdSerialNumber': device_2['cbsdSerialNumber'],
+           'airInterface': device_2['airInterface'],
+           'installationParam': device_2['installationParam'],
+           'measCapability': device_2['measCapability']
+        }
+        # Remove conditionals from registration
+        del device_1['cbsdCategory']
+        del device_1['airInterface']
+        del device_1['installationParam']
+        del device_1['measCapability']
+        del device_2['cbsdCategory']
+        del device_2['airInterface']
+        del device_2['installationParam']
+        del device_2['measCapability']
+       
+        conditionals_iter_1 = {'registrationData': [cbsd_conditional_device_1, cbsd_conditional_device_2]}
+        # Load GWPZ Record
+        gwpz_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+  
+        # Load FSS record
+        fss_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'fss_record_0.json')))
+  
+        # Load ESC record
+        esc_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+     
+        # Load PPA record
+        ppa_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'ppa_record_0.json'))) 
+        pal_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+        
+        # Load DPA record
+        dpa_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'dpa_0.json')))
+        dpa_2 = json.load(
+          open(os.path.join('testcases', 'testdata', 'dpa_1.json')))
+        dpa_3 = json.load(
+          open(os.path.join('testcases', 'testdata', 'dpa_2.json')))
+
+        # Registration configuration for iteration 1
+        cbsd_records_iter_1 = {'registrationRequests': [device_1, device_2],
+                               'grantRequests': [grant_request_1, grant_request_2],
+                               'conditionalRequests': conditionals_iter_1
+                               }
+        cbsd_records_iter_2 = {'registrationRequests': [device_3, device_4],
+                               'grantRequests': [grant_request_3, grant_request_4]}
+        pal_and_ppa_record_1 = {'PAL': pal_record_1, 'PPA': ppa_record_1}
+        protected_entities_iter_1 = {'PalAndPpa': [pal_and_ppa_record_1], 'ESC': [esc_record_1]}
```

Comment:
I think this structure makes the assumption that there is a 1-to-1 correspondence between PAL records and PPAs. In fact, one PPA can be made up of multiple PAL records. Also, a single PAL record can be used to make multiple PPAs. I propose the following structure:

`{
  "palRecords": [pal_record_1],
  "ppas": [ppa_record_1],
  "escs": [esc_record_1],
}
`

Note also the naming style, which matches established conventions.

Reply:
Updated,   as per the comment suggested.